simulation/traffic_simulator/include/traffic_simulator/api/api.hpp

```diff
@@ -74,17 +75,29 @@ class API
       rclcpp::PublisherOptionsWithAllocator<AllocatorT>())),
     debug_marker_pub_(rclcpp::create_publisher<visualization_msgs::msg::MarkerArray>(
       node, "debug_marker", rclcpp::QoS(100), rclcpp::PublisherOptionsWithAllocator<AllocatorT>())),
-    clock_(std::forward<decltype(xs)>(xs)...),
+    clock_(node->get_parameter("use_sim_time").as_bool(), std::forward<decltype(xs)>(xs)...),
     zeromq_client_(
       simulation_interface::protocol, configuration.simulator_host, getZMQSocketPort(*node))
   {
     setVerbose(configuration.verbose);
 
+    real_time_factor_subscriber = rclcpp::create_subscription<std_msgs::msg::Float64>(
+      node, "/real_time_factor", rclcpp::QoS(rclcpp::KeepLast(1)).best_effort(),
+      [this](const std_msgs::msg::Float64 & message) {
+        if (message.data < 0.001) {
```

Comment:
Please explain in the comments why the value of 0.001 is set.

Also, change it to call the client when the condition is met, rather than on an early return (which is my preference).

Additionally, this subscriber does not need to be non-const. Change subscriber to const and initialize it using member initializer list.